### PR TITLE
move build logic for rootless studio into buildkite

### PR DESCRIPTION
**Pull request by elliott-davis**
Signed-off-by: Elliott Davis <elliott@excellent.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: .buildkite/scripts/dockerhub_publish.sh [hunk 1]
```diff
@@ -4,15 +4,27 @@ set -euo pipefail
...
 if is_fake_release; then
   # This overrides IMAGE_NAME in docker-base.sh
   export IMAGE_NAME="habitat/fakey-mc-fake-face-studio"
+else
+  export IMAGE_NAME="habitat/default-studio"
 fi
 
+if [[ -z $DOCKER_LOGIN_USER ]]; then
+  echo "DockerHub user required" && exit 1
+fi
```
Comment (christophermaier) [inline .buildkite/scripts/dockerhub_publish.sh]: Strictly speaking, the `set -u` bit at the top will handle this.

File: .buildkite/scripts/dockerhub_publish.sh [hunk 2]
```diff
@@ -4,15 +4,27 @@ set -euo pipefail
...
+
+if [[ -z $DOCKER_LOGIN_PASSWORD ]]; then
+  echo "DockerHub password required" && exit 1
+fi
+
+docker login -u="$DOCKER_LOGIN_USER" -p="$DOCKER_LOGIN_PASSWORD"
```
Comment (christophermaier) [inline .buildkite/scripts/dockerhub_publish.sh]: Minor, but it's generally good to use the long options in scripts for maximum readability, even though `u` and `p` are probably pretty idiomatic.

```sh
docker login \
       --username="$DOCKER_LOGIN_USER" \
       --password="$DOCKER_LOGIN_PASSWORD"
```

File: .buildkite/scripts/dockerhub_publish.sh [hunk 3]
```diff
@@ -4,15 +4,27 @@ set -euo pipefail
 
 source .buildkite/scripts/shared.sh
 version=$(buildkite-agent meta-data get "version")
-channel=$(buildkite-agent meta-data get release-channel)
 
 if is_fake_release; then
   # This overrides IMAGE_NAME in docker-base.sh
   export IMAGE_NAME="habitat/fakey-mc-fake-face-studio"
+else
+  export IMAGE_NAME="habitat/default-studio"
```
Comment (christophermaier) [inline .buildkite/scripts/dockerhub_publish.sh]: Does this need to be exported now? It looks like you're using it pretty explicitly.

I think the general convention for things that are just local variables (as opposed to environment variables) is to have them be in lower-case, e.g., `image_name`.

It doesn't look like we're sourcing `docker-base.sh`, either, so we can take that comment out.

File: .buildkite/scripts/dockerhub_upload.sh [hunk 4]
```diff
@@ -4,14 +4,30 @@ set -euo pipefail
 
 source .buildkite/scripts/shared.sh
 version=$(buildkite-agent meta-data get "version")
+channel=$(buildkite-agent meta-data get "release-channel")
 
 if is_fake_release; then
   # This overrides IMAGE_NAME in docker-base.sh
   export IMAGE_NAME="habitat/fakey-mc-fake-face-studio"
+else
+  export IMAGE_NAME="habitat/default-studio"
```
Comment (christophermaier) [inline .buildkite/scripts/dockerhub_upload.sh]: Same w/r/t exporting and upper/lower case, and the comment about `docker-base.sh`

File: .buildkite/scripts/dockerhub_upload.sh [hunk 5]
```diff
@@ -4,14 +4,30 @@ set -euo pipefail
...
 fi
 
+if [[ -z $DOCKER_LOGIN_USER ]]; then
+  echo "DockerHub user required" && exit 1
+fi
+
+if [[ -z $DOCKER_LOGIN_PASSWORD ]]; then
+  echo "DockerHub password required" && exit 1
+fi
```
Comment (christophermaier) [inline .buildkite/scripts/dockerhub_upload.sh]: Same w/r/t `set -u`

File: .buildkite/scripts/dockerhub_upload.sh [hunk 6]
```diff
@@ -4,14 +4,30 @@ set -euo pipefail
...
+
+docker login -u="$DOCKER_LOGIN_USER" -p="$DOCKER_LOGIN_PASSWORD"
```
Comment (christophermaier) [inline .buildkite/scripts/dockerhub_upload.sh]: Same w/r/t long options

File: .buildkite/scripts/dockerhub_upload.sh [hunk 7]
```diff
@@ -4,14 +4,30 @@ set -euo pipefail
...
+trap 'rm -f $HOME/.docker/config.json' INT TERM EXIT
+
 pushd ./components/rootless_studio >/dev/null
-./build-docker-image.sh "${version}"
-./tag-docker-image.sh "${version}" "rc"
-./publish-to-dockerhub.sh "${version}" "rc"
+docker build -t habitat:hab-base .
+docker build --build-arg BLDR_CHANNEL="${channel}" --no-cache -t "${IMAGE_NAME}:${version}" "./default"
```
Comment (christophermaier) [inline .buildkite/scripts/dockerhub_upload.sh]: `-t` could be `--tag`.

Also, I don't think you need to quote `"./default"`

Splitting this onto several lines could help the readability, as well.

**Pull request merged**